libs/hbb_common/src/lib.rs

```diff
@@ -321,6 +321,13 @@ pub fn is_ip_str(id: &str) -> bool {
     is_ipv4_str(id) || is_ipv6_str(id)
 }
 
+#[inline]
+pub fn is_hostname_port_str(id: &str) -> bool {
+    regex::Regex::new(r"^[\-.0-9a-zA-Z]+:\d{1,5}$")
```

Comment:
No, this is not correct URL regex.

Reply:
This regex is for hostname, not only url(or domain names).
Hostname like `B3F-W-12345`, `B3F-W-12345.sc` are also accepted here.

May you provide a better regex, or some other requirements?
e.g. `some-string` should be rejected, `some-other-string` should be accepted.

Reply:
URL only please for the time being. for pure hostnane, we may add some prefix.

Reply:
BTW, 1.1.1.1:90 also pass thourgh is_hostname_port_str

Reply:
Regex updated.
And for the prefix, is there any advice?

Reply:
I still think this PR can cause me some trouble in the future. Give me some time.

Reply:
I mean about prefix for domain.

Reply:
I have no good idea also, let's do it later.

Reply:
BTW, intranet hostname actually does not work, our code can not get ip from intranet hostname, we currently use dns lookup. So we do not need to discuss prefix for now.